lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -1355,61 +1359,58 @@ def cmd_loot(*args)
-                      row.push svc
-                    else
-                      row.push ""
-                    end
-                    row.push(loot.ltype)
-                    row.push(loot.name || "")
-                    row.push(loot.content_type)
-                    row.push(loot.info || "")
-                    row.push(loot.path)
-
-                    tbl << row
-                    if (mode == :delete)
-                      loot.destroy
-                      delete_count += 1
-                    end
-                  end
-                end
+            matched_loot_ids = []
+            loots = []
+            if host_ranges.compact.empty?
+              loots = loots + framework.db.loots(framework.db.workspace, {:search_term => search_term})
+            else
+              each_host_range_chunk(host_ranges) do |host_search|
+                loots = loots + framework.db.loots(framework.db.workspace, { :hosts => { :address => host_search }, :search_term => search_term })
               end
             end
 
-            # Handle hostless loot
-            if host_ranges.compact.empty? # Wasn't a host search
-              hostless_loot = framework.db.loots(framework.db.workspace, {:host_id => nil})
-              hostless_loot.each do |loot|
-                row = []
-                row.push("")
-                row.push("")
-                row.push(loot.ltype)
-                row.push(loot.name || "")
-                row.push(loot.content_type)
-                row.push(loot.info || "")
-                row.push(loot.path)
-                tbl << row
-                if (mode == :delete)
-                  loot.destroy
-                  delete_count += 1
+            loots.each do |loot|
+              next if(types and types.index(loot.ltype).nil?)
+              # if search_term
+              #   next unless(
+              #   loot.attribute_names.any? { |a| loot[a.intern].to_s.match(search_term) } or
+              #       loot.host.attribute_names.any? { |a| loot.host[a.intern].to_s.match(search_term) }
+              #   )
+              # end
+              row = []
+              # TODO: This is just a temp implementation of update for the time being since it did not exist before.
+              # It should be updated to not pass all of the attributes attached to the object, only the ones being updated.
+              if mode == :update
+                begin
+                  loot.info = info if info
+                  loot.filename = filename if filename
+                  loot.ltype = types if types
+                  framework.db.update_loot(loot.as_json.symbolize_keys)
+                rescue Exception => e
+                  elog "There was an error updating loot with ID #{loot.id}: #{e.message}"
+                  next
                 end
               end
+              row.push( ((loot.host && loot.host.address) ? loot.host.address : "") )
```

Comment:
Minor: could probably strip out an extra pair of parentheses within the `row.push` call